src/libs/shim-interface/src/lib.rs

```diff
@@ -25,6 +25,47 @@ use kata_types::config::KATA_PATH;
 
 pub const SHIM_MGMT_SOCK_NAME: &str = "shim-monitor.sock";
 
+fn get_uds_with_sid(short_id: &str, kata_run_path: &str) -> Result<String> {
+    // /kata/run/62db246d6c4ffc4a20b56b7259a0346a78aeccb46c4bdc3ef97899d787fd5797
+    let p = Path::new(kata_run_path)
+        .join(short_id)
+        .join(SHIM_MGMT_SOCK_NAME);
+    if p.exists() {
+        return Ok(format!("unix://{}", p.display()));
+    }
+
+    let target_ids: Vec<String> = fs::read_dir(kata_run_path)?
+        .filter_map(|e| {
+            let x = e.ok()?.file_name().to_string_lossy().into_owned();
+            x.as_str().starts_with(short_id).then_some(x)
+        })
+        .collect::<Vec<_>>();
+
+    match target_ids.len() {
+        1 => {
+            // One element and only one exists.
+            Ok(format!(
+                "unix://{}",
+                [kata_run_path, target_ids[0].as_str(), SHIM_MGMT_SOCK_NAME].join("/")
+            ))
+        }
+        n => {
+            // n != 1 return error
+            if n == 0 {
```

Comment:
Nit: It might be clearer to match on `0`, `1` or `n` at the top level.

Reply:
Yeah, both of them are fine for me. but I do as you suggestion